sdk/go/pulumi/types.go

```diff
@@ -389,11 +391,33 @@ func (o *OutputState) ApplyTWithContext(ctx context.Context, applier interface{}
 		}
 
 		// Fulfill the result.
-		result.fulfillValue(results[0], true, nil)
+		result.fulfillValue(results[0], true, secret, nil)
 	}()
 	return result
 }
 
+// IsSecret returns a bool representing the secretness of the Output
+func (o *OutputState) IsSecret() bool {
+	return o.getState().secret
+}
+
+// SecretT wraps the input in an Output marked as secret. The result remains an output
+// property, and accumulates all implicated dependencies, so that resources can be properly tracked using a DAG.
+// This function does not block awaiting the value; instead, it spawns a Goroutine that will await its availability.
+func SecretT(input interface{}) Output {
```

Comment:
This function is morally very similar to `ToOutput`.  Is there a reason this is not `ToSecret` for consistency?

Reply:
Good point. I've updated this.

Reply:
I'll update the programming model docs as well.

Reply:
https://github.com/pulumi/docs/pull/2533